### DISCUSSION TITLE
How to use CreateAction in a Resource

### DISCUSSION THREAD

**Discussion by alessandromarvao** [filamentphp/filament#8621] [Help] [answered]
Hey guys! What's up? I'm developing a system to register products and their purchase orders. In the purchase order Resource, I would like to, in addition to registering orders, also increase the amount of product items in the product table. I created the CreateAction in the resource, but it doesn't work. It only saves the data on the purchase order table, as if the code below wasn't there in the resource.

 Could you help me change the processing of the data when I click on submit the form? This is my code:

`
public static function actions()
{
    return [
        CreateAction::make()
            ->beforeCommit(function ($record, $data) {
                DB::beginTransaction();

                try {
                    Order::create($data);

                    $product = Product::findOrFail($data['product_id']);

                    $product->increment('amount', $data['amount']);

                    DB::commit();
                } catch (\Throwable $th) {
                    DB::rollBack();
                }
            })
            ->redirectTo('orders.index')
            ->flashSuccessMessage('Pedido criado com sucesso.'),
    ];
}
`

Thanks in advance!

**Comment by leandrocfe** [comment, marked as answer]
You can use
```php
->before(function (array $data) {
    //before create
})
->action(function (array $data) {
    //replace the create action
})
->after(function ($record) {
    //after create
})
```

**Reply by alessandromarvao** [reply]
It worked here! Thanks Leandro!
